File: Piece of cake/1207. Unique Number of Occurrences/main.py
#!/usr/bin/env python3
# -*- coding:utf-8 -*-
from typing import *
from collections import Counter, defaultdict
from math import *
from datetime import *
import sys


class Solution:
    def uniqueOccurrences(self, arr: List[int]) -> bool:
        C = Counter(arr)
        C = Counter(list(C.values()))
        return max(list(C.values())) == 1


def main():
    pass


if __name__ == '__main__':
    main()

File: Piece of cake/1207. Unique Number of Occurrences/main.rs
#![allow(dead_code)]
use std::collections::HashMap;
use std::collections::HashSet;
impl Solution {
    pub fn unique_occurrences(arr: Vec<i32>) -> bool {
        let mut counter = HashMap::new();
        arr.into_iter().for_each(|key| { *counter.entry(key).or_insert(0) += 1; });
        let set: HashSet<_> = counter.values().collect();
        set.len() == counter.len()
    }
}

fn main() {}
struct Solution;
